docs/development.md

```diff
@@ -1,6 +1,260 @@
+| suite       | Suite is responsible for parsing the defined formats of different test suites. At the moment only `yaml` is supported but it would be possible to add support for a custom DSL or formats like `toml` and `json`. The `suite.Suite` struct will be used by the `runtime.Runtime` for executing the tests.       | /pkg/suite     |
+
+
+### Build targets
+
+```
+# Initialise dev environment
+$ make init
+
+# Build the project binary
+$ make build
+
+# Unit tests
+$ make test
+
+# Coverage
+$ make test-coverage
+
+# Coverage with more complex tests like ssh execution
+$ make test-coverage-all
+
+# Integration tests for linux and macos
+$ make integration-unix
+
+# Integration on linux
+$ make integration-linux
+
+# Integration windows
+$ make integration-windows
+
+# Add depdencies to vendor
+$ make deps
+```
+
+### Unit tests
+
+`COMMANDER_TEST_ALL` will enable all tests which are depending on external systems like docker or databases.
+Enables ssh tests in unit test suite and sets the credentials for the target host.
+`COMMANDER_SSH_TEST` must be set to `1` to enable ssh tests.
+
+**Note:** I am aware that unit tests should not test external systems nor libraries, but in favour of simplicity and laziness
+I created simple tests inside the directory tree.
+
+```bash
+export COMMANDER_TEST_ALL=1
+export COMMANDER_TEST_SSH=1
+export COMMANDER_TEST_SSH_HOST=localhost:2222
+export COMMANDER_TEST_SSH_PASS=pass
+export COMMANDER_TEST_SSH_USER=root
+export COMMANDER_TEST_SSH_IDENTITY_FILE=integration/containers/ssh/.ssh/id_rsa
+```
+
+## Extending commander
+
+### Add a new field to the `YAML` suite
+
+It is a little bit annoying to add fields because it will be converted multiple times to keep the
+suite format abstracted from the runtime package. 
+The idea behind this it to add support for other formats like `json`, `toml` or maybe a custom DSL.
+
+**Definition of done**:
+
+ - Add a property `message` which always display a message while executing a test
+   ```yaml
+   tests:
+     echo hello:
+       exit-code: 0
+       config:
+         message: this is a very special test
+   ``` 
+ - Support global configurations
+ - Create a simple test case
+
+ **1. Extend the conversion struct `suite.YAMLTestConfigConf` in [yaml_suite.go](../pkg/suite/yaml_suite.go) with the `Message` property.**
+ 
+   The structs types with a `Conf` suffix represent the configuration type, the `YAML` prefix the format of the suite.
+   The naming makes it a little bit clearer in the code which type is used, i.e. a `runtime.CommandUnderTest` or a `suite.YAMLTestConfigConf`.
+    
+    ```go
+    Message    string            `yaml:"message,omitempty"`
+    ```
+    
+**2. Add the `Message` property as a `string` to the `runtime.CommandUnderTest` struct**
+ 
+   The `runtime.CommandUnderTest` struct in [runtime.go](../pkg/runtime/runtime.go) will be used by the runtime to 
+   create the command with all its configs like `env` variables.
+    
```

Comment:
Just to verify that I'm following the struc `CommandUnderTest` would now look like:
```go
type CommandUnderTest struct {
	Cmd        string
	InheritEnv bool
	Env        map[string]string
	Dir        string
	Timeout    string
	Retries    int
	Interval   string
        Message    string
}
```

Reply:
Actually just saw your `7. Learning by doing` (:

Reply:
Yes, that is correct :)